[CI] LeaderCheckerTests testFollowerBehaviour failing

```
java.lang.AssertionError: 	
Expected: a string starting with "[6] consecutive checks of the master node [{leader-1}{LUl9nt39RtSPdM2L9PNAqQ}{0.0.0.0}{0.0.0.0:9}{cdfhimrsw}] were unsuccessful ([1] rejected, [5] timed out), restarting discovery; more details may be available in the master node logs [last unsuccessful check: "	
     but: was "[6] consecutive checks of the master node [{leader-1}{LUl9nt39RtSPdM2L9PNAqQ}{0.0.0.0}{0.0.0.0:9}{cdfhimrsw}] were unsuccessful ([2] rejected, [4] timed out), restarting discovery; more details may be available in the master node logs [last unsuccessful check: simulated error]"	
at __randomizedtesting.SeedInfo.seed([8C3B1DDF41962CA3:C00696D508BE1D48]:0)	
at org.elasticsearch.cluster.coordination.LeaderCheckerTests.lambda$testFollowerBehaviour$1(LeaderCheckerTests.java:170)	
at org.elasticsearch.cluster.coordination.LeaderChecker$CheckScheduler$2.run(LeaderChecker.java:375)	
•••
at org.elasticsearch.cluster.coordination.LeaderCheckerTests.testFollowerBehaviour(LeaderCheckerTests.java:224)	
```

Seems like an assertion message needs to be simply adjusted?

**Build scan:**
https://gradle-enterprise.elastic.co/s/dgjmkegtmtzhy/tests/:server:test/org.elasticsearch.cluster.coordination.LeaderCheckerTests/testFollowerBehaviour

**Reproduction line:**
```
./gradlew ':server:test' --tests "org.elasticsearch.cluster.coordination.LeaderCheckerTests.testFollowerBehaviour" -Dtests.seed=8C3B1DDF41962CA3 -Dtests.locale=ar-EG -Dtests.timezone=Asia/Ust-Nera -Druntime.java=20
```

**Applicable branches:**
7.17

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.cluster.coordination.LeaderCheckerTests&tests.test=testFollowerBehaviour

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: a string starting with "[6] consecutive checks of the master node [{leader-1}{LUl9nt39RtSPdM2L9PNAqQ}{0.0.0.0}{0.0.0.0:9}{cdfhimrsw}] were unsuccessful ([1] rejected, [5] timed out), restarting discovery; more details may be available in the master node logs [last unsuccessful check: "
     but: was "[6] consecutive checks of the master node [{leader-1}{LUl9nt39RtSPdM2L9PNAqQ}{0.0.0.0}{0.0.0.0:9}{cdfhimrsw}] were unsuccessful ([2] rejected, [4] timed out), restarting discovery; more details may be available in the master node logs [last unsuccessful check: simulated error]"

  at __randomizedtesting.SeedInfo.seed([8C3B1DDF41962CA3:C00696D508BE1D48]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.cluster.coordination.LeaderCheckerTests.lambda$testFollowerBehaviour$1(LeaderCheckerTests.java:170)
  at org.elasticsearch.cluster.coordination.LeaderChecker$CheckScheduler$2.run(LeaderChecker.java:375)
  at org.elasticsearch.common.util.concurrent.DeterministicTaskQueue.runTask(DeterministicTaskQueue.java:144)
  at org.elasticsearch.common.util.concurrent.DeterministicTaskQueue.runRandomTask(DeterministicTaskQueue.java:138)
  at org.elasticsearch.common.util.concurrent.DeterministicTaskQueue.runAllRunnableTasks(DeterministicTaskQueue.java:88)
  at org.elasticsearch.cluster.coordination.LeaderCheckerTests.testFollowerBehaviour(LeaderCheckerTests.java:224)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
  at java.lang.reflect.Method.invoke(Method.java:578)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1623)

```

--- a/server/src/test/java/org/elasticsearch/cluster/coordination/LeaderCheckerTests.java
+++ b/server/src/test/java/org/elasticsearch/cluster/coordination/LeaderCheckerTests.java
@@ -115,7 +115,10 @@ public class LeaderCheckerTests extends ESTestCase {
                 super.onSendRequest(requestId, action, request, node);
 
                 final boolean mustSucceed = leaderCheckRetryCount - 1 <= consecutiveFailedRequestsCount;
-                final long responseDelay = randomLongBetween(0, leaderCheckTimeoutMillis + (mustSucceed ? -2 : 60000));
+                final long responseDelay = randomValueOtherThan(
+                    leaderCheckTimeoutMillis,
+                    () -> randomLongBetween(0, leaderCheckTimeoutMillis + (mustSucceed ? -1 : 60000))
+                );
                 final boolean successResponse = allResponsesFail.get() == false && (mustSucceed || randomBoolean());
 
                 assert responseDelay != leaderCheckTimeoutMillis;

--- a/server/src/test/java/org/elasticsearch/cluster/coordination/LeaderCheckerTests.java
+++ b/server/src/test/java/org/elasticsearch/cluster/coordination/LeaderCheckerTests.java
@@ -115,10 +115,11 @@ public class LeaderCheckerTests extends ESTestCase {
                 super.onSendRequest(requestId, action, request, node);
 
                 final boolean mustSucceed = leaderCheckRetryCount - 1 <= consecutiveFailedRequestsCount;
-                final long responseDelay = randomLongBetween(0, leaderCheckTimeoutMillis + (mustSucceed ? -1 : 60000));
+                final long responseDelay = randomLongBetween(0, leaderCheckTimeoutMillis + (mustSucceed ? -2 : 60000));
                 final boolean successResponse = allResponsesFail.get() == false && (mustSucceed || randomBoolean());
 
-                if (responseDelay >= leaderCheckTimeoutMillis) {
+                assert responseDelay != leaderCheckTimeoutMillis;
+                if (responseDelay > leaderCheckTimeoutMillis) {
                     timeoutCount.incrementAndGet();
                     consecutiveFailedRequestsCount += 1;
                 } else if (successResponse == false) {

